### PR TITLE
SemanticSegmentationTrainer: ignore_zeros -> ignore_index

### LINKED ISSUE #444: Bug: missing "not" operator to set ignore_zeros in segmentation trainer
I believe there's a bug in the translation from ignore_zeros to ignore_index in the [\_\_init__ of SemanticSegmentationTask](https://github.com/microsoft/torchgeo/blob/c933694ddc9a121af9415389949aefcc84760472/torchgeo/trainers/segmentation.py#L89).
`self.ignore_zeros = None if kwargs["ignore_zeros"] else 0` should be:
`self.ignore_zeros = None if kwargs["ignore_zeros"] is None else 0` as is done [when define loss function](https://github.com/microsoft/torchgeo/blob/c933694ddc9a121af9415389949aefcc84760472/torchgeo/trainers/segmentation.py#L57)

`self.ignore_zeros` is later used to define the "self.ignore_index" of the self.train_metrics. For sake of clarity, I'd suggest renaming it to  `self.ignore_index` directly in the \_\_init__. 

Therefore, this
```
...
        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0

        self.config_task()

        self.train_metrics = MetricCollection(
            [
                Accuracy(
                    num_classes=self.hparams["num_classes"],
                    ignore_index=self.ignore_zeros,
...
```
would become
```
...
        self.ignore_index = None if kwargs["ignore_zeros"] is None else 0

        self.config_task()

        self.train_metrics = MetricCollection(
            [
                Accuracy(
                    num_classes=self.hparams["num_classes"],
                    ignore_index=self.ignore_index,
...
```
If you'd like, I can definitely push a PR for to fix this bug if the solution seems ok to all. Let me know.

### PR DESCRIPTION
**Pull request by calebrob6**
Fixes #444

### CODE REVIEW

**Comment by adamjstewart** [comment]
Is there any reason not to rename `ignore_zeros` to `ignore_index` in the `__init__` parameter list?

**Comment by calebrob6** [comment]
We don't need to ignore other values and old segmentation trainer checkpoints will not be easily loadable

File: tests/conf/oscd_all.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ experiment:
     in_channels: 26
     num_classes: 2
     num_filters: 1
-    ignore_zeros: True
+    ignore_zeros: False
```
Comment (calebrob6) [inline tests/conf/oscd_all.yaml:14]: We shouldn't ignore_zeros with two classes, this causes the Accuracy metric to break

File: tests/trainers/test_segmentation.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def model_kwargs(self) -> Dict[Any, Any]:
             "encoder_name": "resnet18",
             "encoder_weights": None,
             "in_channels": 1,
-            "num_classes": 1,
+            "num_classes": 2,
```
Comment (calebrob6) [inline tests/trainers/test_segmentation.py:100]: Instantiating the torchmetrics with a single class doesn't make sense

File: torchgeo/trainers/segmentation.py [hunk 3]
```diff
@@ -89,20 +90,30 @@ def __init__(self, **kwargs: Any) -> None:
         self.save_hyperparameters()  # type: ignore[operator]
         self.hyperparams = cast(Dict[str, Any], self.hparams)
 
-        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0
+        if not isinstance(kwargs["ignore_zeros"], bool):
+            raise ValueError("ignore_zeros must be a bool")
+        if kwargs["ignore_zeros"] and kwargs["loss"] == "jaccard":
```
Comment (calebrob6) [inline torchgeo/trainers/segmentation.py:95]: I specifically warn about this because a user might expect training with jaccard to behave similarly to training with ce or focal loss, however it doesn't

**Comment by adamjstewart** [comment]
> We don't need to ignore other values

We don't need to ignore other values _yet_. Presumably this feature exists in torchmetrics because some datasets use a class other than 0 for background. This will break as soon as someone adds a dataset with a different index that needs to be ignored.

> old segmentation trainer checkpoints will not be easily loadable

I'm concerned about reproducibility of experiments, but not necessarily reproducibility with older versions of TorchGeo. We are still in alpha mode and backwards incompatible changes like this would be better sooner rather than later. I still say we should switch to `ignore_index` to match torchmetrics.

**Comment by calebrob6** [comment]
Okay fair enough! Any thoughts on the warning?

File: conf/chesapeake_cvpr.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ experiment:
     in_channels: 4
     num_classes: 7
     num_filters: 256
-    ignore_zeros: False
+    ignore_zeros: null
```
Comment (adamjstewart) [inline conf/chesapeake_cvpr.yaml:20]:
```suggestion
    ignore_index: null
```

File: torchgeo/trainers/segmentation.py [hunk 5]
```diff
@@ -89,20 +89,29 @@ def __init__(self, **kwargs: Any) -> None:
         self.save_hyperparameters()  # type: ignore[operator]
         self.hyperparams = cast(Dict[str, Any], self.hparams)
 
-        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0
-
+        if not (
+            isinstance(kwargs["ignore_index"], int) or (kwargs["ignore_index"] is None)
+        ):
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:94]:
```suggestion
        if not isinstance(kwargs["ignore_index"], (int, type(None))):
```
Another possible option, no preference on which you choose.

File: torchgeo/trainers/segmentation.py [hunk 6]
```diff
@@ -89,20 +89,29 @@ def __init__(self, **kwargs: Any) -> None:
...
+            raise ValueError("ignore_index must be an int or None")
+        if (kwargs["ignore_index"] is not None) and (kwargs["loss"] == "jaccard"):
+            warnings.warn(
+                "Warning ignore_index has no effect on training when"
+                + " loss='jaccard'",
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:99]:
```suggestion
                "ignore_index has no effect on training when loss='jaccard'",
```
The warning will already start with "UserWarning: ..." so no need to warn twice.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]
